File: ios/fingerprint_flutter/Sources/fingerprint_flutter/FingerprintPlugin.swift
import Flutter
import UIKit

public final class FingerprintPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    FingerprintHostApiSetup.setUp(
      binaryMessenger: registrar.messenger(),
      api: FingerprintHostApiImpl()
    )
  }
}

File: android/src/main/kotlin/com/fingerprint/flutter/FingerprintPlugin.kt
package com.fingerprint.flutter

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin

/** Registers the Pigeon [FingerprintHostApi] with the Flutter engine. */
class FingerprintPlugin : FlutterPlugin {
  private var hostApiImpl: FingerprintHostApiImpl? = null

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    hostApiImpl = FingerprintHostApiImpl(flutterPluginBinding.applicationContext)
    FingerprintHostApi.setUp(flutterPluginBinding.binaryMessenger, hostApiImpl)
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    FingerprintHostApi.setUp(binding.binaryMessenger, null)
    hostApiImpl = null
  }
}
